lib/src/migrators/module.dart

```diff
@@ -1228,11 +1252,16 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
   ///
   /// Otherwise, returns [name] unaltered.
   String _unprefix(String name) {
-    var isPrivate = name.startsWith('-');
-    var unprivateName = isPrivate ? name.substring(1) : name;
+    var isPrivate = _isPrivate(name);
+    var unprivateName = isPrivate ? _privateToPublic(name) : name;
     var prefix = _prefixFor(unprivateName);
     if (prefix == null) return name;
-    return (isPrivate ? '-' : '') + unprivateName.substring(prefix.length);
+
+    var withoutPrefix = unprivateName.substring(prefix.length);
+    if (_isPrivate(withoutPrefix)) {
+      withoutPrefix = _privateToPublic(withoutPrefix);
+    }
+    return (isPrivate ? '-' : '') + withoutPrefix;
```

Comment:
```suggestion
    return (isPrivate ? name.substring(0, 1)) + withoutPrefix;
```

Reply:
Generally I'm okay with leaning towards encouraging `-`, since that's definitely the more CSS-style prefix.

Reply:
Yes it's just for the sake of avoiding unecessary changes during a migration, but I definitely don't understand everything that's happening after that, because some tests that look like they should fail with this, don't 🤷 

```scss
<==> arguments
--migrate-deps --remove-prefix=app
<==> input/entrypoint.scss
@import "library";
a {
  color: $app-var;
}

<==> input/_library.scss
$_app-red: red;
$app-var: $_app-red;

<==> output/entrypoint.scss
@use "library";
a {
  color: library.$var;
}

<==> output/_library.scss
$_red: red;
$var: $_red;
```